### ISSUE TITLE
Inconsistent union of literals

### ISSUE THREAD

**Issue by connesy** [python/mypy#14457]
**Bug Report**
A union between literals using the `|` operator yields the error: `error: Unsupported left operand type for | ("object")  [operator]`:

**To Reproduce**
```python
from typing import Literal, Union

A = Literal["A"]
B = Literal["B"]
C = A | B  # Unsupported left operand type for | ("object")
D = Union[A, B]  # No error
```
**Expected Behavior**
I would expect `|` to work exactly the same as `Union`.

**Your Environment**

- Mypy version used: 0.991
- Python version used: 3.10.6

**Comment by connesy** [comment]
This is not an error in Python 3.11.0 :thinking:

**Comment by AlexWaygood** [comment]
This is already fixed on mypy `master`. A fix will be included in the next mypy release, mypy 1.0.

Follow #13685 for updates on the release timeline.

**Comment by connesy** [comment]
Thanks!
